pkg/autogen/v2/autogen.go

```diff
@@ -301,3 +301,46 @@ func (a *ImplAutogenV2) ExtractPodSpec(resource unstructured.Unstructured) (*uns
 
 	return &unstructured.Unstructured{Object: podSpec}, nil
 }
+
+type ExtractPodFunc func(resource *unstructured.Unstructured) (*unstructured.Unstructured, error)
+
+func GetPodExtractor(kind string) ExtractPodFunc {
+	return func(resource *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+		autogen := &ImplAutogenV2{}
+		return autogen.ExtractPodSpec(*resource)
+	}
+}
+
+func ComputeRules(p kyvernov1.PolicyInterface, kind string) ([]kyvernov1.Rule, ExtractPodFunc) {
+	spec := p.GetSpec()
+	applyAutoGen, desiredControllers := CanAutoGen(spec)
+	if !applyAutoGen {
+		desiredControllers = sets.New("none")
+	}
```

Comment:
why are we updating the one of the output of a function based on the other output, this should be inside CanAutoGen func